### ISSUE TITLE
No colors if stdout is redirected

### ISSUE THREAD

**Issue by klenium** [fusesource/jansi#124]
I've write a program that uses both stdout (for "graphical" view) and stderr (for logging), they both have colorized texts. I'm using Win10.
Without any redirect, or if I redirect only stderr, JAnsi works fine.
But if I redirect stdout, the log won't be printed with colors.
This is strange... I think JAnsi uses only stdout for detecting the redirection, but what about stderr?

**Comment by klenium** [comment]
Sorry, I see that it has already been reported.
https://github.com/fusesource/jansi/issues/18
But why is it closed? I'm using the latest (1.17) JAnsi version.

**Comment by klenium** [comment]
Even stranger: if stdout is redirected, then everything that's printed using JAnsi after calling `AnsiConsole.systemUninstall`, it will have colors! If stdout isn't redirected, I see random caracters after the uninstall, just as how it should work.

**Comment by hboutemy** [comment]
System.out and System.err are wrapped independently: see https://github.com/fusesource/jansi/blob/master/jansi/src/main/java/org/fusesource/jansi/AnsiConsole.java#L61
I'm surprised that redirecting stdout affects stderr
You can use {{java -jar jansi*.jar}} to have a diagnostic: please check

**Comment by klenium** [comment]
It works fine on Ubuntu.

For windows 10:
![powershell-diagnostics-normal](https://user-images.githubusercontent.com/4959489/39093693-d8cf1e72-4623-11e8-9967-ab867fc33d54.PNG)
![powershell-diagnostics-redirect](https://user-images.githubusercontent.com/4959489/39093695-e07bafb4-4623-11e8-8d7b-093f4e08bc89.PNG)
![cmd-diagnostics-normal](https://user-images.githubusercontent.com/4959489/39093696-e4c00b6a-4623-11e8-8e35-c59330ca09e0.PNG)
![cmd-diagnostics-redirect](https://user-images.githubusercontent.com/4959489/39093698-e94bce3a-4623-11e8-877f-d7358c1382be.PNG)
![custom-test](https://user-images.githubusercontent.com/4959489/39093700-ec34ea5a-4623-11e8-8e51-97c9cc95d438.PNG)

    public static void main(String[] args) {
        AnsiConsole.systemInstall();
        print(System.out, "Lorem ipsum");
        print(System.err, "dolor sit amet");
        AnsiConsole.systemUninstall();
        print(System.out, "consectetur adipiscing elit");
        print(System.err, "Donec lectus lectus");
    }
    private static void print(PrintStream stream, String text) {
        int half = text.length() / 2;
        stream.print(text.substring(0, half));
        stream.println(Ansi.ansi().fg(Ansi.Color.GREEN).a(text.substring(half)).reset());
    }

**Comment by hboutemy** [comment]
ok, I can reproduce your case with Jansi diagnostic: strange
I had a deeper look: I suppose I should add a new "debug" flag to dump exceptions instead of silently going to fallback, since it's not easy to diagnose
reading comment on https://github.com/fusesource/jansi/blob/master/jansi/src/main/java/org/fusesource/jansi/AnsiConsole.java#L127 and https://github.com/fusesource/jansi/blob/master/jansi/src/main/java/org/fusesource/jansi/WindowsAnsiPrintStream.java#L57 I fear that currently the native code is looking for stdout only, not stderr
we should probably pass some arguments to adapt WindowsAnsiPrintStream to the effective handle (out or err).

need to make some test: help welcome if you can try in parallel

**Comment by hboutemy** [comment]
I finally managed to fix the issue: please test and confirm

**Comment by klenium** [comment]
The first is resolved so stderr has colors if stdout is redirected.

But the uninstall-thing is still wrong. See the last image.

**Comment by hboutemy** [comment]
ok, uninstall is something else: please open another issue

**Comment by hboutemy** [comment]
closing this issue: please open another one for the uninstall case
